source/patterns/@aws-solutions-constructs/aws-wafwebacl-apigateway/test/test.wafwebacl-apigateway.test.ts

```diff
@@ -0,0 +1,361 @@
+        },
+        VisibilityConfig: {
+          CloudWatchMetricsEnabled: true,
+          MetricName: "AWSManagedRulesAnonymousIpList",
+          SampledRequestsEnabled: true
+        }
+      },
+      {
+        Name: "AWS-AWSManagedRulesAmazonIpReputationList",
+        OverrideAction: {
+          None: {}
+        },
+        Priority: 4,
+        Statement: {
+          ManagedRuleGroupStatement: {
+            Name: "AWSManagedRulesAmazonIpReputationList",
+            VendorName: "AWS"
+          }
+        },
+        VisibilityConfig: {
+          CloudWatchMetricsEnabled: true,
+          MetricName: "AWSManagedRulesAmazonIpReputationList",
+          SampledRequestsEnabled: true
+        }
+      },
+      {
+        Name: "AWS-AWSManagedRulesAdminProtectionRuleSet",
+        OverrideAction: {
+          None: {}
+        },
+        Priority: 5,
+        Statement: {
+          ManagedRuleGroupStatement: {
+            Name: "AWSManagedRulesAdminProtectionRuleSet",
+            VendorName: "AWS"
+          }
+        },
+        VisibilityConfig: {
+          CloudWatchMetricsEnabled: true,
+          MetricName: "AWSManagedRulesAdminProtectionRuleSet",
+          SampledRequestsEnabled: true
+        }
+      },
+      {
+        Name: "AWS-AWSManagedRulesSQLiRuleSet",
+        OverrideAction: {
+          None: {}
+        },
+        Priority: 6,
+        Statement: {
+          ManagedRuleGroupStatement: {
+            Name: "AWSManagedRulesSQLiRuleSet",
+            VendorName: "AWS"
+          }
+        },
+        VisibilityConfig: {
+          CloudWatchMetricsEnabled: true,
+          MetricName: "AWSManagedRulesSQLiRuleSet",
+          SampledRequestsEnabled: true
+        }
+      }
+    ]
+  });
+});
+
+// --------------------------------------------------------------
+// Test web acl with user provided acl props
+// --------------------------------------------------------------
+test('Test user provided acl props', () => {
+  const stack = new cdk.Stack();
+  const inProps: lambda.FunctionProps = {
+    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    runtime: lambda.Runtime.NODEJS_10_X,
+    handler: 'index.handler'
+  };
+  const func = defaults.deployLambdaFunction(stack, inProps);
+  const [_api] = defaults.RegionalLambdaRestApi(stack, func);
+  const customRules = [
+    wrapManagedRuleSet("AWSManagedRulesCommonRuleSet", "AWS", 0),
+    wrapManagedRuleSet("AWSManagedRulesWordPressRuleSet", "AWS", 1),
+  ];
+
+  const wafAcl = new waf.CfnWebACL(stack, 'test-waf', {
+    defaultAction: {
+      allow: {}
+    },
+    scope: 'REGIONAL',
+    visibilityConfig: {
+      cloudWatchMetricsEnabled: true,
+      metricName: 'webACL',
+      sampledRequestsEnabled: true
+    },
+    rules: customRules
+  });
+
+  new WafwebaclToApiGateway(stack, 'test-wafwebacl-apigateway', {
+    existingApiGatewayInterface: _api,
+    existingWebaclObj: wafAcl
+  });
+
+  expect(stack).toHaveResource("AWS::WAFv2::WebACL", {
```

Comment:
This is essentially a snapshot test with the snapshot hardcoded in the code. That's going to be a maintenance nightmare. Can we make this more targeted to only check specific values? eg - if set CloudWatchMetricsEnabled: false and then specifically check that value?

Reply:
Check out changes that Hitendra made in [PR 370](https://github.com/awslabs/aws-solutions-constructs/pull/370) for a good example of reducing the check to just the fact being tested and stripping out lots of unneeded CFN. Admittedly Hitendra's is an extreme example... :-)